cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsPathResolver.scala

```diff
@@ -45,17 +46,32 @@ abstract class DrsPathResolver(drsConfig: DrsConfig, retryInternally: Boolean =
   }
 
   private def httpResponseToMarthaResponse(drsPathForDebugging: String)(httpResponse: HttpResponse): IO[MarthaResponse] = {
-    val marthaResponseEntityOption = Option(httpResponse.getEntity).map(EntityUtils.toString)
     val responseStatusLine = httpResponse.getStatusLine
+    val status = responseStatusLine.getStatusCode
 
-    val exceptionMsg = errorMessageFromResponse(drsPathForDebugging, marthaResponseEntityOption, responseStatusLine, drsConfig.marthaUrl)
-    val responseEntityOption = (responseStatusLine.getStatusCode == HttpStatus.SC_OK).valueOrZero(marthaResponseEntityOption)
-    val responseContentIO = toIO(responseEntityOption, exceptionMsg)
+    lazy val retryMessage = {
+      val reason = responseStatusLine.getReasonPhrase
+      s"Unexpected response during resolution of '$drsPathForDebugging': HTTP status $status: $reason"
+    }
 
-    responseContentIO.flatMap{ responseContent =>
-      IO.fromEither(decode[MarthaResponse](responseContent))
-    }.handleErrorWith {
-      e => IO.raiseError(new RuntimeException(s"Unexpected response during DRS resolution: ${ExceptionUtils.getMessage(e)}"))
+    status match {
+      case 408 | 429 =>
+        IO.raiseError(new RuntimeException(retryMessage) with TransientRetryDisposition)
+      case s if s / 100 == 4 =>
+        IO.raiseError(new RuntimeException(retryMessage) with FatalRetryDisposition)
+      case s if s / 100 == 5 =>
+        IO.raiseError(new RuntimeException(retryMessage))
```

Comment:
Should this have a retry disposition like the cases above?

Reply:
I also found this unintuitive, given that these two defined `RetryDisposition` types actually represent cases where we do NOT want to do a "normal" limited-number-of-attempts retry. Maybe make the third one `PersistentRetryDisposition`?

Reply:
I agree that being explicit here is good.

TOL: In other contexts, if I were to choose one of these to be the assumed default, it would probably be `FatalRetryDisposition` because I think `RuntimeException` itself carries that meaning pretty well. However, when considering how `maybeRetryForDownloadFailure` should handle exceptions without any disposition (and based on recent experience with problems stemming from needlessly retrying "bad request" type errors), I'm inclined to agree with keeping the default disposition-less exception handling as written in this PR.